File: 2026-09-11_quantum-ai-brain-frontiers/scripts/build_review.mjs
#!/usr/bin/env node
// Rebuild the bilingual article with Node.js and marked. No network is used by the page.
import fs from 'node:fs';
import path from 'node:path';
import {fileURLToPath} from 'node:url';
const root=path.resolve(path.dirname(fileURLToPath(import.meta.url)),'..');
const moduleRoot=process.env.CODEX_PRIMARY_RUNTIME_NODE_MODULES;
const {marked}=await import(moduleRoot ? path.join(moduleRoot,'marked/lib/marked.esm.js') : 'marked');
const {references,topic_discovery}=JSON.parse(fs.readFileSync(path.join(root,'references.json'),'utf8'));
const escape=s=>String(s).replaceAll('&','&amp;').replaceAll('<','&lt;').replaceAll('>','&gt;').replaceAll('"','&quot;');
const css=`
:root{color-scheme:light;--ink:#183843;--teal:#177b80;--paper:#fbf9f3;--muted:#54676c;--line:#d2dbd8;--gold:#a66c32}
*{box-sizing:border-box}html{scroll-behavior:smooth;scroll-padding-top:92px}
body{margin:0;background:var(--paper);color:var(--ink);font-family:system-ui,-apple-system,BlinkMacSystemFont,"Noto Sans KR",Arial,sans-serif;font-size:18px;line-height:1.94;word-break:keep-all;overflow-wrap:anywhere}
.topbar{position:sticky;top:0;z-index:5;background:rgba(251,249,243,.96);border-bottom:1px solid var(--line);padding:10px max(20px,calc((100vw - 1060px)/2));display:flex;align-items:center;justify-content:space-between;gap:14px;font-size:13px;line-height:1.5}
.topbar a{color:var(--ink);text-decoration:none}.topbar strong{letter-spacing:.11em;font-size:12px}
main{max-width:1000px;margin:0 auto;padding:54px 44px 88px}
h1{font-size:clamp(36px,5vw,61px);font-weight:760;line-height:1.27;letter-spacing:-.04em;margin:18px 0 24px;text-wrap:balance}
.eyebrow,.chapter-no{font-size:12px;line-height:1.4;font-weight:750;letter-spacing:.15em;color:var(--teal)}
.dek{font-size:23px;line-height:1.65;color:var(--muted);max-width:850px;margin:0 0 30px}
h2{font-size:34px;line-height:1.42;letter-spacing:-.025em;margin:105px 0 32px;padding-top:32px;border-top:2px solid var(--ink);text-wrap:balance}
.chapter-no{display:block;margin-bottom:13px}
h3{font-size:24px;line-height:1.5;letter-spacing:-.018em;margin:49px 0 18px}
p{margin:18px 0 24px}a{color:var(--teal);text-underline-offset:4px}strong{font-weight:750}
figure{margin:35px 0 42px}figure img{width:100%;height:auto;display:block;border-radius:12px}
figure.hero{margin:37px -24px 42px}figcaption{font-size:14px;line-height:1.75;color:var(--muted);margin:14px 3px 0}
.scope{border-left:4px solid var(--gold);background:#f2eee3;padding:22px 27px;margin:32px 0;font-size:16px;line-height:1.85}.scope p{margin:8px 0 0}
.contents{display:grid;grid-template-columns:1fr 1fr;gap:0 30px;margin:34px 0 54px;padding:10px 0;border-top:1px solid var(--line);border-bottom:1px solid var(--line)}
.contents a{display:flex;gap:15px;padding:15px 0;font-size:15px;text-decoration:none;line-height:1.6}.contents b{color:var(--gold);min-width:22px}
.equation{font-family:"Cambria Math","STIX Two Math",Georgia,serif;font-size:26px;text-align:center;line-height:1.7;background:#eef2ef;border-radius:10px;padding:25px 22px;margin:28px 0;overflow:auto;white-space:nowrap;overflow-wrap:normal}
.equation.wide{font-size:23px}
.table-wrap{overflow-x:auto;margin:30px 0 38px;border-top:2px solid var(--ink);border-bottom:1px solid var(--line)}
table{border-collapse:collapse;width:100%;min-width:630px;font-size:15px;line-height:1.7}
th,td{text-align:left;vertical-align:top;padding:16px 16px;border-bottom:1px solid var(--line)}th{background:#eaf0ed;font-weight:750}tr:last-child td{border-bottom:0}
ul{padding-left:24px}li{margin:12px 0}.citation{font-size:13px;white-space:nowrap;font-weight:700}
.refs{font-size:14px;line-height:1.75;padding-left:30px}.refs li{padding:9px 0}.refs small{display:block;color:var(--muted);margin-top:3px;font-size:12px}
.topic-links{font-size:14px;line-height:1.8}
.interactive{padding:27px 30px;background:#163944;color:#fff;border-radius:14px;margin:30px 0}
.interactive h4{margin:0 0 8px;font-size:21px}.interactive p{font-size:14px;line-height:1.7;color:#c9dadd;margin:8px 0 20px}
.interactive label{display:block;font-size:16px;margin:12px 0}.interactive input{width:100%;accent-color:#dfaa69;height:30px;cursor:pointer}
.interactive .readout{display:flex;justify-content:space-between;gap:15px;margin:16px 0;font-variant-numeric:tabular-nums;font-size:18px}
.signal-track{height:18px;background:#385762;border-radius:12px;overflow:hidden}.signal-fill{height:100%;background:#efb774;width:1%;transition:width .12s}
button{font:inherit;font-size:14px;border:1px solid #9aafb4;background:transparent;color:white;border-radius:6px;padding:6px 14px;cursor:pointer;margin-top:18px}
button:focus-visible,a:focus-visible,input:focus-visible{outline:3px solid #bf833d;outline-offset:5px}
.language-nav{font-size:14px;text-align:right;margin-bottom:25px}.source-code{font-size:13px;color:var(--muted);border-top:1px solid var(--line);padding-top:24px;margin-top:50px}
@media(max-width:680px){body{font-size:17px;line-height:1.9}main{padding:28px 21px 55px}h1{font-size:37px}.dek{font-size:20px}h2{font-size:28px;margin-top:75px}h3{font-size:22px}figure.hero{margin-left:0;margin-right:0}figcaption{font-size:13px}.contents{grid-template-columns:1fr;gap:0}.contents a{padding:10px 0}.scope{padding:18px 19px}.equation,.equation.wide{font-size:21px;text-align:left}.interactive{padding:22px 20px}.interactive .readout{flex-direction:column;gap:2px}.diagram-wrap{overflow-x:auto;border-radius:12px}.diagram-wrap img{min-width:680px}.topbar strong{font-size:11px}.topbar{font-size:12px}}
@media(prefers-reduced-motion:reduce){html{scroll-behavior:auto}.signal-fill{transition:none}}
@media print{.topbar,.interactive{display:none}body{font-size:11pt;background:white}main{max-width:none;padding:0}h2{break-before:page;margin-top:0}figure,table{break-inside:avoid}a{color:inherit}}
`;
const js=`
(()=>{const slider=document.getElementById('phase');if(!slider)return;
const phase=document.getElementById('phase-value'),prob=document.getElementById('probability'),bar=document.getElementById('signal-fill');
const update=()=>{const v=Number(slider.value),p=Math.sin(v/2)**2;phase.textContent=v.toFixed(2)+' rad';prob.textContent=(100*p).toFixed(3)+'%';bar.style.width=(100*p)+'%';slider.setAttribute('aria-valuetext',v.toFixed(2)+' radians');};
slider.addEventListener('input',update);document.getElementById('phase-reset').addEventListener('click',()=>{slider.value='0.20';update()});update();})();
`;
for(const lang of ['ko','en']){
 let md=fs.readFileSync(path.join(root,'reports',`review.${lang}.md`),'utf8');
 const title=md.match(/^# (.+)$/m)[1];
 const desc=lang==='ko'?'양자진공 복굴절과 LHC의 스핀 상관관계, AI가 공개한 Navier–Stokes 증명의 정확한 범위, 말초 신경 자극과 기억의 근거를 원 논문으로 읽습니다. 개념 일러스트 5장과 도표 6개, 편광 조작 예제로 원리를 설명하고 엘니뇨와 아마존 전망을 짧게 덧붙였습니다.':'A long-form review of vacuum optics, collider quantum information, AI-assisted Navier–Stokes proofs and memory stimulation, with figures and a climate note.';
 const interactive=lang==='ko'?['편광의 위상차를 바꾸어 보기','설명용 위상차 범위입니다. 실제 진공 효과의 크기를 나타내지 않습니다.','상대 위상차 Δφ','교차 방향 검출 비율','초기값으로']:['Change the polarization phase','An enlarged educational range, not the size of a physical vacuum effect.','Relative phase Δφ','Crossed-channel fraction','Reset'];
 md=md.replace('{{POLARIZATION}}',`<section class="interactive" aria-label="${interactive[0]}"><h4>${interactive[0]}</h4><p>${interactive[1]}</p><label for="phase">${interactive[2]}</label><input id="phase" type="range" min="0" max="1.57" value="0.20" step="0.01"><div class="readout"><span>Δφ = <output id="phase-value" for="phase">0.20 rad</output></span><span>${interactive[3]}: <output id="probability" for="phase">0.997%</output></span></div><div class="signal-track" aria-hidden="true"><div class="signal-fill" id="signal-fill"></div></div><button type="button" id="phase-reset">${interactive[4]}</button><noscript><p>Δφ = 0.20 rad → P = sin²(0.10) ≈ 0.997%.</p></noscript></section>`);
 // Link numbered citations before inserting the bibliography.
 md=md.replace(/\[(\d+(?:,\d+)*)\]/g,(_,ids)=>'<span class="citation">'+ids.split(',').map(id=>`<a href="#ref-${id}" aria-label="Reference ${id}">[${id}]</a>`).join(' ')+'</span>');
 md=md.replace('{{REFERENCES}}','<ol class="refs">'+references.map(r=>`<li id="ref-${r.id}"><a href="${r.url}">${escape(r.title)}</a><small>${escape(r.scope)}</small></li>`).join('\n')+'</ol>');
 md=md.replace('{{NEWSLETTER_LINKS}}','<ul class="topic-links">'+topic_discovery.map(r=>`<li><a href="${r.url}">New Scientist · ${escape(r.title)}</a></li>`).join('\n')+'</ul>');
 let body=marked.parse(md,{gfm:true});
 body=body.replace(/<table>/g,'<div class="table-wrap" role="region" tabindex="0" aria-label="'+(lang==='ko'?'비교 표':'Comparison table')+'"><table>').replace(/<\/table>/g,'</table></div>');
 body=body.replace(/(<img[^>]+src="assets\/[^"]+\.svg"[^>]*>)/g,'<div class="diagram-wrap">$1</div>');
 const base='https://infant83.github.io/AI_Tech_Review/reviews/2026-09-11_quantum-ai-brain-frontiers/';
 const canonical=base+(lang==='en'?'en/':'');
 const alternates='<link rel="alternate" hreflang="ko" href="'+base+'"><link rel="alternate" hreflang="en" href="'+base+'en/"><link rel="alternate" hreflang="x-default" href="'+base+'">';
 const languageNav='<nav class="language-nav" aria-label="Language">'+(lang==='ko'?'<span lang="ko" aria-current="page">한국어</span> · <a lang="en" hreflang="en" href="'+base+'en/">English</a>':'<a lang="ko" hreflang="ko" href="'+base+'">한국어</a> · <span lang="en" aria-current="page">English</span>')+'</nav>';
 body=languageNav+body;
 if(lang==='en')fs.cpSync(path.join(root,'dist/assets'),path.join(root,'dist/en/assets'),{recursive:true});
 const html=`<!doctype html><html lang="${lang}"><head><meta charset="utf-8"><meta name="viewport" content="width=device-width,initial-scale=1"><meta name="description" content="${escape(desc)}"><meta name="author" content="AI Tech Review"><meta property="og:type" content="article"><meta property="og:description" content="${escape(desc)}"><meta property="og:url" content="${canonical}"><meta property="og:image" content="${base}vacuum.webp"><meta name="twitter:card" content="summary_large_image">${alternates}<title>${escape(title)}</title><style>${css}</style></head><body><header class="topbar"><a href="https://infant83.github.io/AI_Tech_Review/"><strong>AI TECH REVIEW</strong></a><a href="#vacuum">Quantum</a><a href="#ai">AI</a><a href="#brain">Brain</a><span>2026.09.11</span></header><main><article>${body}</article></main><script>${js}</script></body></html>\n`;
 const dest=path.join(root,'dist',...(lang==='en'?['en']:[]));fs.mkdirSync(dest,{recursive:true});fs.writeFileSync(path.join(dest,'index.html'),html);
 console.log(lang+': '+html.length+' characters, description '+desc.length);
}

File: 2026-09-13_multiphoton-reservoir-classical-light/scripts/build_review.py
"""Build a bilingual static review with deterministic original SVG figures."""
from pathlib import Path
import ast, csv, html, math, re, shutil
import markdown
from bs4 import BeautifulSoup

TOPIC = Path(__file__).resolve().parents[1]
ROOT = TOPIC.parent
SLUG = TOPIC.name
BASE = 'https://infant83.github.io/AI_Tech_Review/'
PAPER = 'https://advanced.onlinelibrary.wiley.com/doi/10.1002/advs.77225'
TITLES = {'ko':'빛의 세기에서 광자 수 분포로: 다광자 저장소 계산을 읽다','en':'From light intensity to photon-count distributions: reading multiphoton reservoir computing'}
DESCS = {'ko':'고전광·편광·OAM·광자 계수의 원리에서 861개 특징, 함수 복원 성능, 32분 취득비용과 고전 시뮬레이션 조건까지 검토합니다.','en':'Classical light, polarization, OAM and photon counting: 861 features, regression evidence, 32-minute acquisition and classical simulation conditions.'}
# Reuse the established editorial layout without executing another package.
template=ast.parse((ROOT/'2026-09-10_cash-qse-classical-quantum-chemistry/scripts/build_review.py').read_text())
CSS=next(ast.literal_eval(node.value) for node in template.body if isinstance(node,ast.Assign) and any(isinstance(t,ast.Name) and t.id=='CSS' for t in node.targets))
CSS+='\n.final-article img{border-radius:4px}.final-article ul{padding-left:24px}.final-article li{font-size:15px}.kicker{font-size:13px;letter-spacing:.1em;color:#5f6b65;margin-bottom:18px}.standfirst{max-width:780px;font-size:18px;line-height:1.8;color:#5f6b65;margin:22px 0 0}.final-article p:has(>img[src="pipeline.svg"]),.final-article p:has(>img[src="poisson_features.svg"]){border:1px solid #d8d5cb;background:white}.math{display:inline-block;vertical-align:middle}.final-article li .math{display:inline}\n'

def svg_start(w,h,title):
    return [f'<svg xmlns="http://www.w3.org/2000/svg" width="{w}" height="{h}" viewBox="0 0 {w} {h}" role="img"><title>{html.escape(title)}</title><rect width="100%" height="100%" fill="#ffffff"/><g font-family="Arial, Noto Sans KR, sans-serif" fill="#17211d">']

def text(out,x,y,value,size=17,color='#17211d',anchor='start'):
    out.append(f'<text x="{x}" y="{y}" font-size="{size}" fill="{color}" text-anchor="{anchor}">{html.escape(value)}</text>')

def pipeline(lang):
    ko=lang=='ko'; out=svg_start(760,510,'Optical feature acquisition and trained readout')
    text(out,32,39,'무엇을 고정하고 무엇을 학습하나' if ko else 'What stays fixed, and what is trained',22)
    labels=[('입력 · 편광','91개 입력 상태','INPUT') if ko else ('Input · polarization','91 input settings','INPUT'),('고정 · 광학 모드 혼합','파장판 + q-plate','FIXED') if ko else ('Fixed · optical mixing','Wave plates + q-plates','FIXED'),('측정 · 계수 분포','21 OAM × 41 항목 = 861','MEASURE') if ko else ('Measured · count features','21 OAM × 41 entries = 861','MEASURE'),('학습 · 고전 선형 읽기층','가중치 + ridge 정규화','TRAINED') if ko else ('Trained · classical readout','Weights + ridge penalty','TRAINED')]
    positions=[(32,65),(402,65),(32,248),(402,248)]
    for (x,y),(a,b,c) in zip(positions,labels):
        out.append(f'<rect x="{x}" y="{y}" width="326" height="132" rx="7" fill="{("#e8f2ed" if c=="TRAINED" else "#f2f5f6")}" stroke="#cad4d1"/>')
        text(out,x+20,y+29,c,12,'#0f766e');text(out,x+20,y+65,a,19);text(out,x+20,y+99,b,16,'#5f6b65')
    out.append('<defs><marker id="a" viewBox="0 0 10 10" refX="9" refY="5" markerWidth="7" markerHeight="7" orient="auto-start-reverse"><path d="M 0 0 L 10 5 L 0 10 z" fill="#0f766e"/></marker></defs><path d="M365 130H394 M565 203V222H195V240 M365 315H394" fill="none" stroke="#0f766e" stroke-width="2" marker-end="url(#a)"/>')
    text(out,32,427,'측정 시간은 빛의 전파 시간과 별도로 필요합니다.' if ko else 'Acquisition is a separate cost from optical propagation.',18)
    text(out,32,460,'1초 / 입력·OAM 설정 → 전체 1,911초 ≈ 32분' if ko else '1 s / input–OAM setting → 1,911 s ≈ 32 min total',17,'#25456f')
    text(out,32,488,'구성 변경·교정·데이터 이동 시간 제외' if ko else 'Excludes switching, calibration and data transfer',14,'#5f6b65')
    return ''.join(out)+ '</g></svg>'

def poisson(lang):
    ko=lang=='ko';out=svg_start(760,490,'Illustrative ideal Poisson feature map')
    text(out,32,37,'광량 하나에서 서로 다른 비선형 특징으로' if ko else 'Different nonlinear features from one intensity',22)
    text(out,32,67,'이상적 코히런트 모드 · 실험 데이터가 아닌 수식 예제' if ko else 'Ideal coherent mode · mathematical illustration, not measured data',14,'#5f6b65')
    x0,y0,w,h=80,110,610,285
    for p in [0,.25,.5,.75,1]:
        y=y0+h*(1-p);out.append(f'<path d="M{x0} {y}H{x0+w}" stroke="#e0e5e3"/>');text(out,x0-12,y+5,f'{p:g}',13,'#5f6b65','end')
    for lam in range(9):
        x=x0+w*lam/8;text(out,x,y0+h+24,str(lam),13,'#5f6b65','middle')
    for n,color in [(0,'#25456f'),(1,'#0f766e'),(2,'#bc6414'),(4,'#925681')]:
        points=[]
        for i in range(401):
            lam=8*i/400;p=math.exp(-lam)*lam**n/math.factorial(n)
            points.append(f'{x0+w*lam/8:.2f},{y0+h*(1-p):.2f}')
        out.append(f'<polyline points="{" ".join(points)}" fill="none" stroke="{color}" stroke-width="3"/>')
    for i,(n,color) in enumerate([(0,'#25456f'),(1,'#0f766e'),(2,'#bc6414'),(4,'#925681')]):
        x=150+i*142;out.append(f'<path d="M{x} 91h22" stroke="{color}" stroke-width="3"/>');text(out,x+30,96,f'n = {n}',14)
    text(out,385,450,'평균 검출 광자 수 λ' if ko else 'Mean detected photon number λ',16,'#17211d','middle')
    text(out,385,480,'Pₙ(λ) = exp(−λ) λⁿ / n!     ·     n = 0, 1, 2, 4',15,'#5f6b65','middle')
    text(out,32,99,'확률' if ko else 'Probability',13,'#5f6b65')
    return ''.join(out)+'</g></svg>'

def render_md(source):
    saved=[]
    def protect(m):
        raw=m.group(0);display=raw.startswith('\\[')
        saved.append('<span class="'+('display-math' if display else 'math')+'">'+html.escape(raw)+'</span>')
        return f'MATHBLOCKTOKEN{len(saved)-1}END'
    body=markdown.markdown(re.sub(r'\\\[.*?\\\]|\\\(.*?\\\)',protect,source,flags=re.S),extensions=['tables','fenced_code','sane_lists'])
    for i,v in enumerate(saved):body=body.replace(f'MATHBLOCKTOKEN{i}END',v)
    return body

def main():
    hero=TOPIC/'artifacts/photon_reservoir_hero.webp'
    if not hero.exists():raise FileNotFoundError(hero)
    rows=[['target','mean_intensity_R2','count_distribution_R2','source'],*[ [f,a,b,'Hong et al. 2026 SI Table S2'] for f,a,b in [('quadratic',.92,.94),('cubic',.89,.93),('quartic',.90,.94),('exponential',.96,.97),('logarithmic',.97,.97),('damped_sinusoid',.69,.94),('reported_mean',.89,.95)]]]
    for lang in ['ko','en']:
        ko=lang=='ko';dist=TOPIC/'dist'/('' if ko else 'en');dist.mkdir(parents=True,exist_ok=True)
        shutil.copy2(hero,dist/hero.name)
        (dist/'pipeline.svg').write_text(pipeline(lang));(dist/'poisson_features.svg').write_text(poisson(lang))
        with (dist/'benchmarks.csv').open('w',newline='',encoding='utf-8') as f:csv.writer(f,lineterminator='\n').writerows(rows)
        source=(TOPIC/'reports'/('final_review.md' if ko else 'final_review_en.md')).read_text()
        (dist/'review.md').write_text(source)
        csvurl=f'https://github.com/Infant83/AI_Tech_Review/blob/main/{SLUG}/dist/{"" if ko else "en/"}benchmarks.csv'
        soup=BeautifulSoup(render_md(source.split('\n',1)[1].replace('(benchmarks.csv)',f'({csvurl})')),'html.parser')
        toc=[]
        for i,h in enumerate(soup.find_all('h2'),1):
            h['id']=f'section-{i}';toc.append(f'<a href="#section-{i}">{html.escape(h.get_text())}</a>')
        for table in soup.find_all('table'):
            table.wrap(soup.new_tag('div',attrs={'class':'table-wrap','tabindex':'0','role':'region','aria-label':'연구 비교 표' if ko else 'Research comparison table'}))
        for im in soup.find_all('img'):
            name=im['src'];im['width']='1536' if name==hero.name else '760';im['height']='1024' if name==hero.name else ('510' if name=='pipeline.svg' else '490')
            im['fetchpriority']='high' if name==hero.name else 'auto'
            if name!=hero.name:im['loading']='lazy'
        url=BASE+'reviews/'+SLUG+('/' if ko else '/en/');other=BASE+'reviews/'+SLUG+('/en/' if ko else '/')
        title=html.escape(TITLES[lang]);desc=html.escape(DESCS[lang]);mdurl=f'https://github.com/Infant83/AI_Tech_Review/blob/main/{SLUG}/dist/{"" if ko else "en/"}review.md'
        full=f'''<!doctype html><html lang="{lang}"><head><meta charset="utf-8"><meta name="viewport" content="width=device-width, initial-scale=1"><title>{title}</title><meta name="description" content="{desc}"><link rel="canonical" href="{url}"><link rel="alternate" hreflang="ko" href="{BASE}reviews/{SLUG}/"><link rel="alternate" hreflang="en" href="{BASE}reviews/{SLUG}/en/"><link rel="alternate" hreflang="x-default" href="{BASE}reviews/{SLUG}/"><meta property="og:type" content="article"><meta property="og:title" content="{title}"><meta property="og:description" content="{desc}"><meta property="og:url" content="{url}"><meta property="og:image" content="{url}{hero.name}"><meta name="twitter:card" content="summary_large_image"><meta name="author" content="김현중"><meta property="article:published_time" content="2026-09-13"><meta property="article:modified_time" content="2026-09-13"><style>{CSS}</style>
<script>window.MathJax={{tex:{{inlineMath:[['\\\\(','\\\\)']],displayMath:[['\\\\[','\\\\]']]}},options:{{skipHtmlTags:['script','noscript','style','textarea','pre','code']}}}};</script><script defer src="https://cdn.jsdelivr.net/npm/mathjax@3.2.2/es5/tex-chtml.js" integrity="sha384-AHAnt9ZhGeHIrydA1Kp1L7FN+2UosbF7RQg6C+9Is/a7kDpQ1684C2iH2VWil6r4" crossorigin="anonymous"></script></head><body><div class="topline"><div class="topline-inner"><span class="brand">AI Tech Review Letters</span><div class="topline-actions"><a href="{BASE}">{'리뷰 허브' if ko else 'Review hub'}</a><span aria-current="page">{'한국어' if ko else 'English'}</span><a href="{other}" hreflang="{'en' if ko else 'ko'}">{'English' if ko else '한국어'}</a><a href="{mdurl}">Markdown</a></div></div></div><header class="hero"><div class="hero-inner"><div class="kicker">PHOTONIC COMPUTING · 2026.09.13</div><h1>{title}</h1><p class="standfirst">{desc}</p></div></header><main class="content-grid"><article class="final-article">{soup}<div class="publication-note">{'생성 표지는 개념 일러스트입니다. 두 SVG 도표는 논문의 측정 흐름과 이상적 Poisson 식을 설명하기 위해 제작했습니다.' if ko else 'The generated hero is conceptual. Two original SVG figures explain the measurement workflow and ideal Poisson equation.'}</div></article><aside class="sidebar"><h2>{'이 글의 순서' if ko else 'In this article'}</h2><nav aria-label="{'목차' if ko else 'Contents'}">{''.join(toc)}</nav><div class="side-extra"><hr><a href="{PAPER}">Advanced Science · e77225</a><a href="{csvurl}">{'표 데이터 CSV' if ko else 'Table data CSV'}</a><p>{'동료평가 논문 · 상온 광학 실험' if ko else 'Peer-reviewed · room-temperature optics'}</p></div></aside></main></body></html>'''
        full=full.replace('<span aria-current="page">',f'<span lang="{lang}" aria-current="page">')
        (dist/'index.html').write_text(full)
        print(lang,len(source.split()),'words',len(source),'characters')

if __name__=='__main__':main()
